ext/session/session.c

```diff
@@ -783,6 +783,34 @@ static PHP_INI_MH(OnUpdateSidBits) /* {{{ */
 }
 /* }}} */
 
+static PHP_INI_MH(OnUpdateSessionGcProbability) /* {{{ */
+{
+    SESSION_CHECK_ACTIVE_STATE;
+    SESSION_CHECK_OUTPUT_STATE;
+
+    if (atol(ZSTR_VAL(new_value)) < 0) {
+        php_error_docref(NULL, E_WARNING, "session.gc_probability cannot be negative");
+        return FAILURE;
+    }
+
+    return OnUpdateLong(entry, new_value, mh_arg1, mh_arg2, mh_arg3, stage);
+}
+/* }}} */
+
+static PHP_INI_MH(OnUpdateSessionDivisor) /* {{{ */
+{
+    SESSION_CHECK_ACTIVE_STATE;
+    SESSION_CHECK_OUTPUT_STATE;
+
+    if (atol(ZSTR_VAL(new_value)) <= 0) {
+        php_error_docref(NULL, E_WARNING, "session.gc_divisor must be greater than 0");
+        return FAILURE;
+    }
+
```

Comment:
Please use `zend_ini_parse_quantity_warn()` like in the implementation of `ZEND_INI_MH(OnUpdateLongGEZero)`

Reply:
Thanks. It's done